oeps/oep-0011.rst

```diff
@@ -0,0 +1,412 @@
+   and modern front end libraries typically use ES2015 (aka ES6).
+   Most browsers now provide native support for ES2015, but to support
+   ES2016 or older browsers it is recommended to use `Babel`_ which can
+   transpile various JavaScript dialects into ES5 (or ES2015, in the future).
+
+   Note: use of ES2015 features should be preferred to the use of older
+   libraries such as CoffeeScript and Underscore which have been subsumed
+   into the newer standard.
+
+3. **JavaScript libraries should be installed via npm**
+
+   Rationale: It is important that JavaScript libraries are kept
+   up-to-date, and one key component is to make it as simple as possible
+   to perform upgrades. Projects that use npm maintain all of their
+   library requirements in a single package.json file, which can be
+   easily changed as the versions change.
+
+4. **JavaScript code should follow the edX ESLint configuration**
+
+   Rationale: In order to standardize and enforce Open edX's JavaScript
+   coding style across multiple codebases, edX has published an ESLint
+   configuration that provides an enforceable specification. The
+   `edX ESLint Config`_ is made available as an npm package that can be
+   installed into any Open edX package.
+
+5. **Use React and Redux when building new client-side applications**
+
+   Rationale: There are a large number of JavaScript libraries for
+   building rich client applications, but many of them are too
+   rigid for a flexible, extensible codebase like Open edX. The
+   edX front end working group (aka FedX) has chosen the combination
+   of React and Redux as the optimal way to build new client-side
+   applications. A great primer on the benefits of this approach is
+   `A Study Plan to Cure JavaScript Fatigue`_.
+
+   An interesting area for future exploration is `React Native`_ which
+   is a variant of React that can be used to render native iOS and Android
+   applications. This could be a potential solution to build consistent
+   experiences more efficiently across web and mobile. See `React at edX`_
+   for more details.
+
+   Note: the other modern library that many organizations are using
+   is Angular 2.  See `Reasons for rejecting Angular`_ for an exploration
+   as to why this does not work as well as React for edX.
+
+   Exception: when building new applications within edx-platform, it
+   is currently acceptable to use Backbone.  Backbone is a somewhat old
+   technology in the JavaScript world and has seen a rapid drop off in usage
+   within the community. The plan is to incrementally adopt React within
+   edx-platform as it can interoperate cleanly with Backbone.
+
+6. **JavaScript dependencies should be managed with AMD modules**
+
+   Rationale: Open edX is a large, complicated code base, and it is
+   getting more tangled over time. It is important that applications be
+   composed as modules, and AMD is the standard module system for JavaScript.
```

Comment:
> AMD is the standard module system for JavaScript

Eeeeeehhhhhhh..... ES2015 modules are closer to being "the standard" given that they're officially endorsed by ECMA and browsers plan to implement them natively. AMD was, at the time that we picked it, a widely adopted standard, but to say that it is "the standard system" isn't 100% accurate IMO.